dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -0,0 +1,239 @@
+import {useDispatch} from 'react-redux'
+import "./index.css";
+import {
+  ToggleGroup,
+  ToggleGroupItem,
+  Page,
+  Masthead,
+  MastheadToggle,
+  PageSidebar,
+  PageToggleButton,
+  PageSection,
+  PageSectionVariants,
+} from "@patternfly/react-core";
+import {
+  TableComposable,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+} from "@patternfly/react-table";
+import axios from "axios";
+import BarsIcon from "@patternfly/react-icons/dist/js/icons/bars-icon";
+import SearchBox from "../SearchComponent";
+import DatePickerWidget from "../DatePickerComponent";
+import Heading from "../HeadingComponent";
+import PathBreadCrumb from "../BreadCrumbComponent";
+import AlertMessage from "../AlertComponent";
+import NavItems from "../NavbarComponent";
+import EmptyTable from "../EmptyStateComponent";
+import moment from "moment";
+import { fetchPublicDatasets } from "../../../actions/fetchPublicDatasets";
+let startDate = moment(new Date(1990, 10, 4)).format("YYYY/MM/DD");
+let endDate = moment(new Date(2040, 10, 4)).format("YYYY/MM/DD");
+let controllerName = "";
+let dataArray = [];
+export const TableWithFavorite = () => {
+  const columnNames = {
+    controller: "Controller",
+    name: "Name",
+    creationDate: "Created On",
+  };
+  const [activeSortIndex, setActiveSortIndex] = useState(null);
+  const [activeSortDirection, setActiveSortDirection] = useState(null);
+  const [favoriteRepoNames, setFavoriteRepoNames] = useState([]);
+  const [publicData, setPublicData] = useState([]);
+  const [isSelected, setIsSelected] = useState("firstButton");
+  const [isNavOpen, setIsNavOpen] = useState(false);
+  const dispatch=useDispatch();
+  useEffect(() => {
+     dispatch(fetchPublicDatasets()).then((res) => {
+        dataArray = res.data;
+        setPublicData(res.data);
+        setFavoriteRepoNames(JSON.parse(localStorage.getItem("favControllers")))
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, []);
+  const markRepoFavorited = (repo, isFavoriting = true) =>{
+   const otherFavorites = favoriteRepoNames.filter((r) => r.name !== repo.name);
+      if (isFavoriting) saveFavorites([...otherFavorites, repo]);
+      else saveFavorites(otherFavorites);
+      const newFavorite= isFavoriting ? [...otherFavorites, repo] : otherFavorites;
+      setFavoriteRepoNames(newFavorite);
+  }
+    // setFavoriteRepoNames((prevFavorites) => {
+    //   const otherFavorites = prevFavorites.filter((r) => r.name !== repo.name);
+    //   if (isFavoriting) saveFavorites([...otherFavorites, repo]);
+    //   else saveFavorites(otherFavorites);
+    //   return isFavoriting ? [...otherFavorites, repo] : otherFavorites;
+    // }
+
+  const onNavToggle = () => {
+    setIsNavOpen(!isNavOpen);
+  };
+  const isRepoFavorited = (repo) => {
+    for (let i = 0; i < favoriteRepoNames.length; i++) {
+      if (
+        repo.name === favoriteRepoNames[i].name &&
+        repo.controller === favoriteRepoNames[i].controller &&
```

Comment:
Dataset names are unique regardless of controller, so it's not necessary to match against both.

(A lot of the old APIs do require both, but I removed the controller from `datasets/list`, `datasets/publish`, and `datasets/delete` deliberately, and we'll continue moving towards "demoting" controller entirely to just another piece of metadata nobody needs to see.)